README.md

```diff
@@ -305,6 +306,50 @@ The default `interval` is `50ms`. However it will run your callback immediately
 on the next tick of the event loop (in a `setTimeout`) before starting the
 intervals.
 
+### `waitForElements`
+
+Defined as:
+
+```typescript
+function waitForElements<T>(
+  callback?: () => T | null | undefined,
+  options?: {
+    container?: HTMLElement
+    timeout?: number
+    mutationObserverOptions?: MutationObserverInit
+  },
+): Promise<T>
+```
+
+When in need to wait for DOM elements to appear, disappear, or change you can use `waitForElements`.
+The `waitForElements` function is a small wrapper
+around the
+[`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).
+Here's a simple example:
+
+```javascript
+// ...
+// wait until the callback does not throw an error and returns a truthy value. In this case, that means
+// it'll wait until we can get a form control with a label that matches "username"
+// the difference from `wait` is that it reacts to DOM changes in the container
+// and returns the value returned by the callback
+const element = await waitForElements(() => getByLabelText(container, 'username'))
+element.value = 'chucknorris'
+// ...
+```
+
+Using `MutationObserver` is more efficient than polling the DOM at regular intervals with `wait`.
```

Comment:
Let's add a link to the MDN docs for this so folks know that it's actually a browser feature. Most people have probably never heard of it. I only heard of it a few weeks ago when building the cypress command waiter.